core/consensus/babe/impl/babe_config_repository_impl.cpp

```diff
@@ -58,17 +83,32 @@ namespace kagome::consensus::babe {
   }
 
   bool BabeConfigRepositoryImpl::prepare() {
-    auto load_res = load();
-    auto best_info = block_tree_->bestLeaf();
-    auto best_block = block_tree_->getBlockHeader(best_info.hash).value();
-    if ((not load_res or not config({best_info}, 0))
-        and trie_storage_->getEphemeralBatchAt(best_block.state_root)) {
-      readFromState(best_info);
-      load_res = outcome::success();
+    auto last_indexed = indexer_.removeUnfinalized();
+    auto finalized = block_tree_->getLastFinalized();
+    auto finalized_header = block_tree_->getBlockHeader(finalized.hash).value();
+    if (finalized.number - last_indexed.number > kDontIndexFinalizedBlocks
+        and trie_storage_->getEphemeralBatchAt(finalized_header.state_root)) {
```

Comment:
What do you want to check with this expression?

Reply:
If last finalized has state (full sync, completed fast or warp sync) and there are a lot of unindexed blocks, then don't index them and read last state from runtime.

Reply:
getEphemeralBatchAt may return an error even if the state is present, for example if a database error occurred. If you want to check if the state is present, compare it with DatabaseError::NOT_FOUND

Reply:
That's ok, because `getEphemeralBatchAt` must return success to successfully call runtime